forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,329 @@
+    {
+        println!("Deleting {:?}", snapshot_path);
+        if !yes && !prompt_confirm() {
+            println!("Aborted.");
+            return;
+        }
+
+        delete_snapshot(&snapshot_path);
+    } else {
+        println!(
+                "{:?} is not a valid snapshot file path, to list all snapshots, run forest-cli snapshot list",
+                snapshot_path);
+    }
+}
+
+fn prune(config: &Config, snapshot_dir: &Option<PathBuf>, yes: bool) {
+    {
+        let snapshot_dir = snapshot_dir
+            .clone()
+            .unwrap_or_else(|| default_snapshot_dir(config));
+        println!("Snapshot dir: {:?}", snapshot_dir);
+        let mut snapshots_with_valid_name = vec![];
+        let mut snapshot_to_keep = None;
+        let mut latest_date = Date::MIN;
+        let mut latest_height = 0;
+        let pattern = Regex::new(
+            r"^forest_snapshot_([^_]+?)_(?P<date>\d{4}-\d{2}-\d{2})_height_(?P<height>\d+).car$",
+        )
+        .unwrap();
+        if let Ok(dir) = fs::read_dir(snapshot_dir) {
+            println!("\nLocal snapshots:");
+            for path in dir
+                .flatten()
+                .map(|entry| entry.path())
+                .filter(|p| p.is_file())
+            {
+                if let Some(Some(filename)) = path.file_name().map(|n| n.to_str()) {
+                    let captures = pattern.captures(filename);
+                    if let Some(captures) = captures {
+                        if let Some(date) = captures.name("date") {
+                            if let Ok(date) = time::Date::parse(date.as_str(), &Iso8601::DEFAULT) {
+                                if let Some(height) = captures.name("height") {
+                                    if let Ok(height) = height.as_str().parse::<i64>() {
+                                        if date > latest_date {
+                                            latest_date = date;
+                                            latest_height = height;
+                                            snapshot_to_keep = Some(path.clone());
+                                        } else if date == latest_date && height > latest_height {
+                                            latest_height = height;
+                                            snapshot_to_keep = Some(path.clone());
+                                        }
+
+                                        snapshots_with_valid_name.push(path);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if snapshots_with_valid_name.len() < 2 {
+            println!("No files to delete");
+            return;
+        }
+
+        let mut snapshots_to_delete = snapshots_with_valid_name;
+        let mut index_to_keep = 0;
+        println!("Files to delete:");
+        if let Some(snapshot_to_keep) = snapshot_to_keep {
```

Comment:
could we somehow simplify this with https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.position ?

Reply:
It does `println!` in the for loop, I think it's better not to call functions with a side effect in the lambda

Reply:
Ah yes, I see.